eb_hooks.py

```diff
@@ -756,64 +756,141 @@ def post_postproc_cuda(self, *args, **kwargs):
-                    basename = filename.split('.')[0]
-                    if basename in allowlist:
-                        self.log.debug("%s is found in allowlist, so keeping it: %s", basename, full_path)
-                    else:
-                        self.log.debug("%s is not found in allowlist, so replacing it with symlink: %s",
-                                       basename, full_path)
-                        # if it is not in the allowlist, delete the file and create a symlink to host_injections
-
-                        # the host_injections path is under a fixed repo/location for CUDA
-                        host_inj_path = re.sub(EESSI_INSTALLATION_REGEX, HOST_INJECTIONS_LOCATION, full_path)
-                        # CUDA itself doesn't care about compute capability so remove this duplication from
-                        # under host_injections (symlink to a single CUDA installation for all compute
-                        # capabilities)
-                        accel_subdir = os.getenv("EESSI_ACCELERATOR_TARGET")
-                        if accel_subdir:
-                            host_inj_path = host_inj_path.replace("/accel/%s" % accel_subdir, '')
-                        # make sure source and target of symlink are not the same
-                        if full_path == host_inj_path:
-                            raise EasyBuildError("Source (%s) and target (%s) are the same location, are you sure you "
-                                                 "are using this hook for an EESSI installation?",
-                                                 full_path, host_inj_path)
-                        remove_file(full_path)
-                        symlink(host_inj_path, full_path)
+        # replace files that are not distributable with symlinks into
+        # host_injections
+        replace_non_distributable_files_with_symlinks(self.log, self.installdir, self.name, allowlist)
     else:
         raise EasyBuildError("CUDA-specific hook triggered for non-CUDA easyconfig?!")
 
 
+def post_postproc_cudnn(self, *args, **kwargs):
+    """
+    Remove files from cuDNN installation that we are not allowed to ship,
+    and replace them with a symlink to a corresponding installation under host_injections.
+    """
+
+    # We need to check if we are doing an EESSI-distributed installation
+    eessi_installation = bool(re.search(EESSI_INSTALLATION_REGEX, self.installdir))
+
+    if self.name == 'cuDNN' and eessi_installation:
+        print_msg("Replacing files in cuDNN installation that we can not ship with symlinks to host_injections...")
+
+        allowlist = ['LICENSE']
+
+        # read cuDNN LICENSE, construct allowlist based on section "2. Distribution" that specifies list of files that can be shipped
+        license_path = os.path.join(self.installdir, 'LICENSE')
+        search_string = "2. Distribution. The following portions of the SDK are distributable under the Agreement:"
```

Comment:
The sanity check doesn't help because we need to install the targets of the symlinks.

We can raise an exception if the line is not found. Probably good to be cautious. ... another option might be to only proceed with the processing if the LICENSE file is the same that we have based the code on here? We could calculate the hash of it and require it to be a specific one.

Reply:
Added raising an error if search string wasn't found in https://github.com/EESSI/software-layer/pull/772/commits/fc00d0cf758e6f4b250cbbd3e33a2db53f046740